website/static/js/statistics.js

```diff
@@ -0,0 +1,242 @@
+                .chartType('bar')
+                .chartOptions({
+                    tooltip:{
+                        format:{
+                            name: function(){return 'Visits';}
+                        }
+                    },
+                    axis: {
+                        x: {
+                            label: {
+                                text: 'Hour of Day',
+                                position: 'outer-center',
+                            },
+                            tick: {
+                                centered: true,
+                                values: ['0', '4', '8', '12', '16', '20'],
+                            },
+                        },
+                        y: {
+                            tick: {
+                                format: self._hideNonIntegers,
+                            }
+                        }
+                    },
+                });
+
+        // make sure all hours of the day 0-23 are present
+        var munger = function() {
+            var foundHours = {};
+            for (var i=this.dataset.matrix.length-1; i>0; i--) {
+                var row = this.dataset.selectRow(i);
+                foundHours[ row[0] ] = row[1];
+                this.dataset.deleteRow(i);
+            }
+
+            for (var hour=0; hour<24; hour++) {
+                var stringyNum = '' + hour;
+                this.dataset.appendRow(stringyNum, [ foundHours[stringyNum] || 0 ]);
+            }
+        };
+        self.buildChart(query, dataviz, munger);
+
+    };
+
+    // most popular sub-pages of this project
+    self.popularPages = function(elem, startDate, endDate) {
+        var query = {
+            type: 'count_unique',
+            params: {
+                event_collection: 'pageviews',
+                timeframe: {
+                    start: startDate.toISOString(),
+                    end: endDate.toISOString(),
+                },
+                target_property: 'anon.id',
+                group_by: 'page.title'
+            }
+        };
+
+        var dataviz = new keenDataviz()
+                .el(elem)
+                .chartType('bar')
+                .chartOptions({
+                    tooltip:{
+                        format:{
+                            name: function(){return 'Visits';}
+                        }
+                    },
+                    axis: {
+                        y: {
+                            tick: {
+                                format: self._hideNonIntegers,
+                            }
+                        }
+                    },
+                });
+
+        var munger = function() {
+            this.dataset.updateColumn(0, function(value, index, column) {
+                var title = value.replace(/^OSF \| /, '');
+                // Strip off the project title, if present at beginning of string
+                if (title.startsWith(window.contextVars.node.title)) {
```

Comment:
We generally try to avoid using contextVars in re-usable modules. Can this information be passed in as arguments?

This isn't a blocker.